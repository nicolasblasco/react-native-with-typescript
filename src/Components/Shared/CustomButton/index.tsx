import React from 'react';
import {Text, StyleSheet, Pressable} from 'react-native';

interface Props {
  onPress: any;
  text: string;
  testID?: string;
}

const CustomButton: React.FC<Props> = ({onPress, text, testID}) => {
  return (
    <>
      <Pressable onPress={onPress} style={styles.container} testID={testID}>
        <Text style={styles.text}>{text}</Text>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    margin: 20,
    padding: 15,
    borderRadius: 5,
    backgroundColor: '#19456B',
  },
  text: {
    fontWeight: 'bold',
    color: '#FFFFFF',
  },
});

export default CustomButton;
