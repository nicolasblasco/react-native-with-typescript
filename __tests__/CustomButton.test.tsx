import 'react-native';
import React from 'react';
import CustomButton from '../src/Components/shared/CustomButton';

import {render, waitFor} from '@testing-library/react-native';

describe('CustomButton', () => {
  it('Renders correctly', async () => {
    const {getByText, queryByTestId, toJSON} = render(
      <CustomButton
        onPress={() => {}}
        text="Test text"
        testID="example-button"
      />,
    );

    await waitFor(() => expect(queryByTestId('example-button')).toBeTruthy());

    expect(getByText('Test text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
