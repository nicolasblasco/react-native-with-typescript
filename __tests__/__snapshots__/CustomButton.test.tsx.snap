// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#19456B",
      "borderRadius": 5,
      "margin": 20,
      "padding": 15,
    }
  }
  testID="example-button"
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontWeight": "bold",
      }
    }
  >
    Test text
  </Text>
</View>
`;
